BUILD.md

```diff
@@ -0,0 +1,44 @@
+# Prerequisite
+```
+export CLANG_PREFIX=$($HOME/clang+llvm-tapir5.0/bin/llvm-config --prefix);
+export LLVM_SOURCES-/tmp/llvm_sources-tapir5.0;
+cd ${LLVM_SOURCES};
+git clone --recursive https://github.com/wsmoses/Tapir-LLVM llvm && \
+    mkdir -p ${LLVM_SOURCES}/llvm_build && cd ${LLVM_SOURCES}/llvm_build && \
+    ${CMAKE_VERSION} -DLLVM_ENABLE_EH=ON -DLLVM_ENABLE_OCAMLDOC=OFF -DLLVM_INSTALL_OCAMLDOC_HTML_DIR=/tmp -DLLVM_OCAML_INSTALL_PATH=/tmp -DCMAKE_INSTALL_PREFIX=${CLANG_PREFIX} -DLLVM_TARGETS_TO_BUILD=X86 -DCOMPILER_RT_BUILD_CILKTOOLS=OFF -DLLVM_ENABLE_CXX1Y=ON -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_BUILD_TESTS=OFF -DLLVM_ENABLE_ASSERTIONS=ON -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_LLVM_DYLIB=ON  -DLLVM_ENABLE_RTTI=ON ../llvm/ && \
+    make -j"$(nproc)" -s && make install -j"$(nproc)" -s
+```
+
+# Conda from scratch (one time config)
+```
+wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh && \
+    chmod +x miniconda.sh && \
+    ./miniconda.sh -b -p ~/conda/miniconda && \
+    rm miniconda.sh
+
+conda create -y --name tc_build python=3.6
+conda activate tc_build
+conda install -y gflags glog gtest protobuf
+conda install -y pyyaml mkl-include
+conda install -yc conda-forge pytest
+conda install pytorch torchvision cuda90 -c pytorch
+```
+
+# Activate preinstalled conda in your current terminal
+```
+. ~/conda/miniconda/bin/activate
+condo activate tc_build
+```
+
+# Build TC
+The following assumes a predefined location for `CUDA_TOOLKIT_ROOT_DIR`, `CUDNN_ROOT_DIR` and `CLANG_PREFIX`, adapt to your needs.
+```
+CUDA_TOOLKIT_ROOT_DIR=/public/apps/cuda/9.0/ CUDNN_ROOT_DIR=/public/apps/cudnn/v7.0/cuda/ CLANG_PREFIX=$($HOME/clang+llvm-tapir5.0/bin/llvm-config --prefix) ./build.sh --all
```

Comment:
Shouldn't `CUDA_TOOLKIT_ROOT_DIR` come from conda?  We have installed cuda-9.0 with it.

Reply:
We have not installed cuda-9.0 with conda, it's just an option for installing pytorch.
That being said it does indeed pull some dependencies but only libraries (no headers and no binaries).